h2o-core/src/main/java/water/parser/CsvParser.java

```diff
@@ -110,6 +110,7 @@
             break;
           }
           if ((!isEOL(c) || quoteCount == 1) && ((quotes != 0) || (c != CHAR_SEPARATOR))) {
+            if (str.getBuffer() == null) str.set(bits, offset, 0);
```

Comment:
This seems suspicious here. We should initialize `str` when we enter the `STRING` state. This kind of handling here is indicative that there might be an issue somewhere else in the code.

Can you please explain?

Reply:
From that it seems to me that we should initialized the stringbuffer when we hit a new line and we know we are in quotes, or am I missing something?

Reply:
This is exactly what we do. Only the if could hide some unwanted state that would otherwise end with a failure, you're right.